### DISCUSSION TITLE
AIO Container problem due to full disk

### DISCUSSION THREAD

**Discussion by freakyy85** [nextcloud/all-in-one#1218] [Questions] [answered]
Hello!

After I had a full SSD (no diskspace left) im getting the following error when going to mydomain:8443 (the aio interface):

> Slim Application Error
> The application could not run because of the following error:
> 
> Details
> Type: TypeError
> Code: 0
> Message: AIO\Data\ConfigurationManager::GetConfig(): Return value must be of type array, null returned
> File: /var/www/docker-aio/php/src/Data/ConfigurationManager.php
> Line: 15
> Trace
> #0 /var/www/docker-aio/php/src/Data/ConfigurationManager.php(455): AIO\Data\ConfigurationManager->GetConfig()
> #1 /var/www/docker-aio/php/src/Data/ConfigurationManager.php(425): AIO\Data\ConfigurationManager->GetEnvironmentalVariableOrConfig('APACHE_PORT', 'apache_port', '443')
> #2 /var/www/docker-aio/php/src/ContainerDefinitionFetcher.php(81): AIO\Data\ConfigurationManager->GetApachePort()
> #3 /var/www/docker-aio/php/src/ContainerDefinitionFetcher.php(201): AIO\ContainerDefinitionFetcher->GetDefinition(true)
> #4 /var/www/docker-aio/php/src/ContainerDefinitionFetcher.php(32): AIO\ContainerDefinitionFetcher->FetchDefinition()
> #5 /var/www/docker-aio/php/src/Docker/DockerActionManager.php(719): AIO\ContainerDefinitionFetcher->GetContainerById('nextcloud-aio-a...')
> #6 /var/www/docker-aio/php/public/index.php(115): AIO\Docker\DockerActionManager->isLoginAllowed()
> #7 /var/www/docker-aio/php/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(43): Closure->{closure}(Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
> #8 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(384): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(GuzzleHttp\Psr7\ServerRequest), Object(GuzzleHttp\Psr7\Response), Array)
> #9 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\Routing\Route->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #10 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #11 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/Route.php(341): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #12 /var/www/docker-aio/php/vendor/slim/slim/Slim/Routing/RouteRunner.php(84): Slim\Routing\Route->run(Object(GuzzleHttp\Psr7\ServerRequest))
> #13 /var/www/docker-aio/php/vendor/slim/csrf/src/Guard.php(456): Slim\Routing\RouteRunner->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #14 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(209): Slim\Csrf\Guard->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Slim\Routing\RouteRunner))
> #15 /var/www/docker-aio/php/vendor/slim/twig-view/src/TwigMiddleware.php(115): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #16 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\Views\TwigMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
> #17 /var/www/docker-aio/php/src/Middleware/AuthMiddleware.php(38): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #18 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(313): AIO\Middleware\AuthMiddleware->__invoke(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
> #19 /var/www/docker-aio/php/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(107): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #20 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(147): Slim\Middleware\ErrorMiddleware->process(Object(GuzzleHttp\Psr7\ServerRequest), Object(Psr\Http\Server\RequestHandlerInterface@anonymous))
> #21 /var/www/docker-aio/php/vendor/slim/slim/Slim/MiddlewareDispatcher.php(81): Psr\Http\Server\RequestHandlerInterface@anonymous->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #22 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(215): Slim\MiddlewareDispatcher->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #23 /var/www/docker-aio/php/vendor/slim/slim/Slim/App.php(199): Slim\App->handle(Object(GuzzleHttp\Psr7\ServerRequest))
> #24 /var/www/docker-aio/php/public/index.php(164): Slim\App->run()
> #25 {main}
> [Go Back](https://cloud.nextgamers.eu:8443/login#)`

Is there a chance of recovering this? :)

**Comment by szaimen** [comment]
Yes, you can restore the configuration.json from a backup archive using https://github.com/nextcloud/all-in-one#pro-tip-backup-archives-access

But I see that we probably need to add some logic that prevents something like this from happening.

**Reply by szaimen** [reply]
This is now fixed with v2.0.4 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by freakyy85** [comment]
hm, it says:

root@s:~# sudo mkdir -p /tmp/borg && sudo borg mount "/mnt/backup/borg" /tmp/borg
Repository /mnt/backup/borg does not exist.
root@s:~#

**Reply by freakyy85** [reply]
Sorry I don't know why this does strike-through.

**Reply by freakyy85** [reply]
I think this is running in a container? oO

**Reply by szaimen** [reply]
If you have no backup, there is no recover to this.

**Reply by szaimen** [reply]
However it is possible that the backup repository is not stored in the /mnt/backup/borg path as you can specify the path on the host yourself.

**Reply by szaimen** [reply]
What is the output of `sudo ls var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/` ?

**Reply by freakyy85** [reply]
root@s:~# sudo ls /var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/
backup_archives.list  backupsecret  borg.config  configuration.json  daily_backup_time

**Reply by szaimen** [reply]
Okay good. So a backup was created.

**Reply by szaimen** [reply]
Now you need to find the directory where it is stored on the host system.

**Reply by freakyy85** [reply]
its all standard?

**Reply by szaimen** [reply]
Are you sure? Then the folder `/mnt/backup/borg` should exist.

**Comment by freakyy85** [comment]
root@s:~# updatedb
root@s:~# locate configuration.json
/var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/configuration.json
root@s:~#

**Comment by szaimen** [comment, marked as answer]
Please run `sudo find / -type d -name "borg"` in order to find the folder

**Reply by freakyy85** [reply]
root@s:/mnt# sudo find / -type d -name "borg"
/tmp/borg
/var/backups/borg
/var/lib/docker/overlay2/5305663e190bd3a321fff5d0bf117fdcaf291a27fde843ef84d81f908e0a55bc/diff/usr/lib/python3/dist-packages/borg
/var/lib/docker/volumes/nextcloud_aio_backup_cache/_data/.config/borg
/var/lib/docker/volumes/nextcloud_aio_backup_cache/_data/.cache/borg
/usr/lib/python3/dist-packages/borg
root@s:/mnt#

it's /var/backups/borg then?

**Reply by szaimen** [reply]
> it's /var/backups/borg then?

Yes

**Reply by freakyy85** [reply]
ok i mounted it to /tmp/borg now how do i restore the config?

**Reply by freakyy85** [reply]
root@s:/tmp/borg/20221001_040115-nextcloud-aio# ls
nextcloud_aio_volumes
root@s:/tmp/borg/20221001_040115-nextcloud-aio#

**Reply by szaimen** [reply]
Inside this archive and inside the mastercontainer volume shiuld be a configuration.json. copy that back to `var/lib/docker/volumes/nextcloud_aio_mastercontainer/_data/data/` and restart the mastercontainer

**Reply by freakyy85** [reply]
very nice. thank you very much it worked :)
